### PR TITLE
split vm tests

### PR DESCRIPTION
**Pull request by winsvega**
tests https://github.com/ethereum/tests/pull/296

### CODE REVIEW

**Comment by chfast** [comment]
Can you explain what this is about?

File: test/tools/jsontests/vm.cpp [hunk 1]
```diff
@@ -480,114 +474,55 @@ json_spirit::mValue doVMTestsNoLog(json_spirit::mValue const& _input, bool _fill
...
-	dev::test::executeTests("vmBitwiseLogicOperationTest", "/VMTests", "/VMTestsFiller", dev::test::doVMTests);
-}
-
-BOOST_AUTO_TEST_CASE(vmSha3Test)
-{
-	dev::test::executeTests("vmSha3Test", "/VMTests", "/VMTestsFiller", dev::test::doVMTests);
-}
 
-BOOST_AUTO_TEST_CASE(vmEnvironmentalInfoTest)
+class vmtestfixture
```
Comment (chfast) [inline test/tools/jsontests/vm.cpp]: I have no idea, why out of nowhere you named this class this way.

File: test/tools/jsontests/vm.cpp [hunk 2]
```diff
@@ -480,114 +474,55 @@ json_spirit::mValue doVMTestsNoLog(json_spirit::mValue const& _input, bool _fill
...
 		{
-			BOOST_ERROR(" Failed test with Exception: " << diagnostic_information(_e));
+			cnote << "Skipping " << casename << " because --all option is not specified.\n";
+			return;
 		}
-		catch (std::exception const& _e)
+		fillAllFilesInFolder(casename);
+	}
+
+	void fillAllFilesInFolder(string const& _folder)
```
Comment (pirapira) [inline test/tools/jsontests/vm.cpp]: Maybe you can make this `private`.

File: test/tools/jsontests/vm.cpp (hunk 1 above)
Comment (winsvega) [reply test/tools/jsontests/vm.cpp]: following the same scheme as for the other tests.

File: test/tools/jsontests/vm.cpp (hunk 2 above)
Comment (winsvega) [reply test/tools/jsontests/vm.cpp]: I am going to remove this function in the next PR

File: test/tools/jsontests/vm.cpp (hunk 1 above)
Comment (chfast) [reply test/tools/jsontests/vm.cpp]: Like `ConstantinopleTransitionTestFixture` or `TestOutputFixture`?

Comment (winsvega) [reply test/tools/jsontests/vm.cpp]: like transactiontestfixture, generaltestfixture

Comment (pirapira) [reply test/tools/jsontests/vm.cpp]: Let's do CamelCase in class names.

**Comment by winsvega** [comment]
this PR is about refactoring the testeth. it moves towards creation of a function
runAllTestsInFolder that will unify the tests execution for all suites

**Comment by winsvega** [comment]
https://github.com/ethereum/cpp-ethereum/pull/4383

**Comment by chfast** [comment]
I don't think  a single test case "run all tests in a folder" is a good idea if you register the test names first (to be available to Boost.Test filtering).

**Comment by winsvega** [comment]
the test cases will be as it is right now

**Comment by pirapira** [comment]
I want to understand what @winsvega is trying to do.  I might like it, I might not like it, but it's better to know before an actual implementation happens.

Will `runAllTestsInFolder()` be used in within a test case or out of a test case?

Will `runAllTestsInFolder()` invoke multiple test cases?

Will `runAllTestsInFolder()` called from test cases, from test fixtures, or from somewhere else?

**Comment by winsvega** [comment]
@pirapira we discussed this on skype. 
runAllTestsInFolder will run from fixtures to basically generalize the logic that we have in all suites. 
and to make check of all files in folders at one place not repeating the code in different fixtures
runAllTestsInFolder invoce 1 test case but used in many test cases

**Comment by pirapira** [comment]
@winsvega sounds good.

**Pull request merged**